README.txt

After the installation, set the Page ID (pid) of the product list plugin 
of commerce (pi1). Just put the extension on a site of your choice and 
select a category, where the search should start.

The pagebrowser is also configurable through the flex forms.

The localization of the pagebrowser can be made through the 
_LOCAL_LANG.[your language key] directive in the typoscript configuration. 
(See below for an example)

You can also activate the plugin through typoscript:

lib.mycontent = COA
lib.mycontent{
  10 < plugin.tx_commercesearch_pi1
}

after that, there is an other subpart in the templatefile (formular_ts), so 
a different display is possible.


EXAMPLE TypoScript configuration
--------------------------------

plugin.tx_commercesearch_pi1{
 #size from teaser and images of the product
 file{
 	maxH = 40
 	maxW = 40
 }

 # this is need if you want to display the formular on other page as the plugin is placed.
 # specify where the output of the result should be 
 targetPage = 48
 
 # startcategory for searching
 # selects also the subcategories and itself
 startCategory = 2

 # LOCALIZATION
 _LOCAL_LANG.de{
    pi_list_browseresults_first = erste
    pi_list_browseresults_prev = << vorige
    pi_list_browseresults_page = Seite
    pi_list_browseresults_next = nächste >>
    pi_list_browseresults_last = letzte
    pi_list_browseresults_displays = Zeige Einträge %s bis %s von insgesamt %s
 }
}


EXAMPLE CSS Configuration
-------------------------

/*commerce search*/
.tx-commercesearch-pi1{text-align:left;}
.tx-commercesearch-pi1 a{color:black;}

.tx-commercesearch-pi1-browsebox p{}


.tx_commercesearch_pi1_letternavigation{background:url(../gfx/bg_table_header.gif); height: 22px; text-align: center;}
.tx_commercesearch_pi1_letternavigation span{padding-left: 3px; padding-right: 3px;}
.tx_commercesearch_pi1_letternavigation a{color: white; text-transform: uppercase; font-weight: bold;}


table.tx_commercesearch_pi1_result_table{
	border-collapse:collapse;
	border-spacing:0px;
	margin-bottom:8px;
	width:100%;
}

table.tx_commercesearch_pi1_result_table thead{
  background-image:url(../gfx/bg_table_header.gif);
	height: 22px;
}

table.tx_commercesearch_pi1_result_table th {color: white;}
table.tx_commercesearch_pi1_result_table th.image{width: 40px;}
table.tx_commercesearch_pi1_result_table td {}
table.tx_commercesearch_pi1_result_table tr.even {
  background-image:url(../gfx/bg_table_even.gif);
	height: 22px;
}

table.tx_commercesearch_pi1_result_table tr.odd {
  color: #6c2925 !important;
  background-image:url(../gfx/bg_table_odd.gif);
	height: 22px;
}


.tx-commercesearch-pi1{}
.tx-commercesearch-pi1 table a:link, .tx-commercesearch-pi1 table a:active, .tx-commercesearch-pi1 table a:visited {
  text-decoration: none; /*color: #602020*/ color: black;}
  
.tx-commercesearch-pi1 table a:hover {text-decoration: underline; font-weight:bold; color: black;}

.tx-commercesearch-pi1 table th{background-image: none;}
.tx-commercesearch-pi1 table.tx_commercesearch_pi1_formular_table label{}
.tx-commercesearch-pi1 table.tx_commercesearch_pi1_formular_table input{width: 140px;}
.tx-commercesearch-pi1 table.tx_commercesearch_pi1_formular_table select{width: 144px;}

#tx_commercesearch_pi1_formular_ts form{margin: 0; padding: 0;}
.tx-commercesearch-pi1 table.tx_commercesearch_pi1_formular_table_small{margin: 0px; padding:0px;}
.tx-commercesearch-pi1 table.tx_commercesearch_pi1_formular_table_small label{font-weight: normal;}
.tx-commercesearch-pi1 table.tx_commercesearch_pi1_formular_table_small input{width: 105px;}
.tx-commercesearch-pi1 table.tx_commercesearch_pi1_formular_table_small select{width: 100%;}


ChangeLog
(add new changes on top of this file)

09-07-08 Andreas Jonderko  <typo3@alnovi.de>
  * Search in products and articles (title subtitle plain_text)
  * Added search types (default, exact, startswith)


13-01-08 Andreas Jonderko  <typo3@alnovi.de>

  * Flexform has template choice
	* Rendering through templates
	* Image Display
	* TS Configuration of file, target page for display 
  * Localization
  * SQL Optimizing

11-01-08 Andreas Jonderko  <typo3@alnovi.de>

	* Initial release


ext_tables.php
<?php
if (!defined ('TYPO3_MODE')) 	die ('Access denied.');

t3lib_div::loadTCA('tt_content');
$TCA['tt_content']['types']['list']['subtypes_excludelist'][$_EXTKEY.'_pi1']='layout,select_key';
$TCA['tt_content']['types']['list']['subtypes_addlist'][$_EXTKEY.'_pi1']='pi_flexform';    // flexform

t3lib_extMgm::addPlugin(array('LLL:EXT:commerce_search/locallang_db.xml:tt_content.list_type_pi1', $_EXTKEY.'_pi1'),'list_type');

t3lib_extMgm::addStaticFile($_EXTKEY,"pi1/static/","Commerce Search");

t3lib_extMgm::addPiFlexFormValue($_EXTKEY.'_pi1', 'FILE:EXT:commerce_search/flexform.xml');

if (TYPO3_MODE=="BE")	$TBE_MODULES_EXT["xMOD_db_new_content_el"]["addElClasses"]["tx_commercesearch_pi1_wizicon"] = t3lib_extMgm::extPath($_EXTKEY).'pi1/class.tx_commercesearch_pi1_wizicon.php';

t3lib_extMgm::addStaticFile($_EXTKEY,'static/static/', 'static');
?>

pi1/class.tx_commercesearch_pi1_wizicon.php
<?php
/***************************************************************
*  Copyright notice
*
*  (c) 2008 Andreas Jonderko <anj2j@yahoo.de>
*  All rights reserved
*
*  This script is part of the TYPO3 project. The TYPO3 project is
*  free software; you can redistribute it and/or modify
*  it under the terms of the GNU General Public License as published by
*  the Free Software Foundation; either version 2 of the License, or
*  (at your option) any later version.
*
*  The GNU General Public License can be found at
*  http://www.gnu.org/copyleft/gpl.html.
*
*  This script is distributed in the hope that it will be useful,
*  but WITHOUT ANY WARRANTY; without even the implied warranty of
*  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
*  GNU General Public License for more details.
*
*  This copyright notice MUST APPEAR in all copies of the script!
***************************************************************/




/**
 * Class that adds the wizard icon.
 *
 * @author	Andreas Jonderko <typo3@pixelmotor.de>
 * @package	TYPO3
 * @subpackage	tx_commercesearch
 */
class tx_commercesearch_pi1_wizicon {

					/**
					 * Processing the wizard items array
					 *
					 * @param	array		$wizardItems: The wizard items
					 * @return	Modified array with wizard items
					 */
					function proc($wizardItems)	{
						global $LANG;

						$LL = $this->includeLocalLang();

						$wizardItems['plugins_tx_commercesearch_pi1'] = array(
							'icon'=>t3lib_extMgm::extRelPath('commerce_search').'pi1/ce_wiz.gif',
							'title'=>$LANG->getLLL('pi1_title',$LL),
							'description'=>$LANG->getLLL('pi1_plus_wiz_description',$LL),
							'params'=>'&defVals[tt_content][CType]=list&defVals[tt_content][list_type]=commerce_search_pi1'
						);

						return $wizardItems;
					}

					/**
					 * Reads the [extDir]/locallang.xml and returns the $LOCAL_LANG array found in that file.
					 *
					 * @return	The array with language labels
					 */
					function includeLocalLang()	{
						$llFile = t3lib_extMgm::extPath('commerce_search').'locallang.xml';
						$LOCAL_LANG = t3lib_div::readLLXMLfile($llFile, $GLOBALS['LANG']->lang);
						
						return $LOCAL_LANG;
					}
				}



if (defined('TYPO3_MODE') && $TYPO3_CONF_VARS[TYPO3_MODE]['XCLASS']['ext/commerce_search/pi1/class.tx_commercesearch_pi1_wizicon.php'])	{
	include_once($TYPO3_CONF_VARS[TYPO3_MODE]['XCLASS']['ext/commerce_search/pi1/class.tx_commercesearch_pi1_wizicon.php']);
}

?>

ext_emconf.php
<?php

########################################################################
# Extension Manager/Repository config file for ext: "commerce_search"
#
# Auto generated 09-07-2008 21:05
#
# Manual updates:
# Only the data in the array - anything else is removed by next write.
# "version" and "dependencies" must not be touched!
########################################################################

$EM_CONF[$_EXTKEY] = array(
	'title' => 'Commerce Search',
	'description' => 'Searchform for the Commerce Shop Extension, selectable category, result is displayed with a pagebrowser, extension is configurable through flexforms, display through template. See Readme.txt for some informations.',
	'category' => 'plugin',
	'author' => 'Andreas Jonderko',
	'author_email' => 'typo3 { a  t } alnovi.de',
	'shy' => '',
	'dependencies' => 'commerce',
	'conflicts' => '',
	'priority' => '',
	'module' => '',
	'state' => 'beta',
	'internal' => '',
	'uploadfolder' => 0,
	'createDirs' => '',
	'modify_tables' => '',
	'clearCacheOnLoad' => 0,
	'lockType' => '',
	'author_company' => 'Alnovi',
	'version' => '0.2.1',
	'constraints' => array(
		'depends' => array(
			'commerce' => '',
		),
		'conflicts' => array(
		),
		'suggests' => array(
		),
	),
	'_md5_values_when_last_written' => 'a:20:{s:9:"ChangeLog";s:4:"013a";s:10:"README.txt";s:4:"57a8";s:21:"ext_conf_template.txt";s:4:"0f64";s:12:"ext_icon.gif";s:4:"32c3";s:17:"ext_localconf.php";s:4:"20c0";s:14:"ext_tables.php";s:4:"0c07";s:12:"flexform.xml";s:4:"3ec8";s:13:"locallang.xml";s:4:"6564";s:16:"locallang_db.xml";s:4:"8333";s:27:"static/static/constants.txt";s:4:"956b";s:23:"static/static/setup.txt";s:4:"f531";s:14:"pi1/ce_wiz.gif";s:4:"32c3";s:35:"pi1/class.tx_commercesearch_pi1.php";s:4:"4a77";s:43:"pi1/class.tx_commercesearch_pi1_wizicon.php";s:4:"c01d";s:13:"pi1/clear.gif";s:4:"cc11";s:17:"pi1/locallang.xml";s:4:"f15d";s:30:"pi1/tx_commercesearch_pi1.html";s:4:"4713";s:24:"pi1/static/editorcfg.txt";s:4:"14d1";s:19:"doc/wizard_form.dat";s:4:"4495";s:20:"doc/wizard_form.html";s:4:"ab54";}',
	'suggests' => array(
	),
);

?>